ethereumj-core/src/main/java/org/ethereum/publish/LifeCycleSubscription.java

```diff
@@ -0,0 +1,53 @@
+package org.ethereum.publish;
+
+import org.ethereum.publish.event.Event;
+
+import java.util.function.BiConsumer;
+
+/**
+ * Abstraction that holds together event type, event processor (a.k.a consumer) and optional conditions that resolves
+ * event processing and auto-unsubscribe.
+ *
+ * @author Eugene Shevchenko
+ */
+public class LifeCycleSubscription<E extends Event<D>, D> extends Subscription<E, D> {
```

Comment:
I bet we don't need yet another abstraction here. Introducing it makes an API less obvious for users. I am more about to have two constructors for `Subscription`, one with `BiConsumer`, another one with `Consumer` that is extended to `BiConsumer` during construction.

Reply:
Ok, I'll move this functionality to `Subscription`